extensions/replication/queueProcessor/QueueProcessor.js

```diff
@@ -354,97 +425,150 @@ class QueueProcessor {
+                    this._setTargetAccountMd(destEntry, targetRole, log,
+                                             next);
                 },
                 // put metadata in target bucket
-                (accountAttr, next) => {
+                next => {
                     // TODO check that bucket role matches role in metadata
-                    this._putMetaData('target', destEntry, log, next);
+                    this._putMetadata('target', destEntry, log, next);
                 },
-            ], _writeReplicationStatus);
+            ], _handleReplicationOutcome);
         }
         return async.waterfall([
             // get data stream from source bucket
             next => {
-                this._getBucketReplicationRoles(sourceEntry, log, next);
+                this._setupRoles(sourceEntry, log, next);
             },
             (sourceRole, targetRole, next) => {
-                this._processRoles(sourceEntry, sourceRole,
-                                   destEntry, targetRole, log, next);
+                this._setTargetAccountMd(destEntry, targetRole, log, next);
             },
             // Get data from source bucket and put it on the target bucket
-            (accountAttr, next) => {
-                // TODO check that bucket role matches role in metadata
+            next => {
                 this._getAndPutData(sourceEntry, destEntry, log, next);
             },
             // update location, replication status and put metadata in
             // target bucket
             (location, next) => {
                 destEntry.setLocation(location);
-                this._putMetaData('target', destEntry, log, next);
+                this._putMetadata('target', destEntry, log, next);
             },
-        ], _writeReplicationStatus);
+        ], _handleReplicationOutcome);
+    }
+
+    _tryUpdateReplicationStatus(updatedSourceEntry, params, done) {
+        const { backoffCtx, log, reason } = params;
+        const _doneUpdate = err => {
+            if (!err) {
+                log.info('replication status updated',
+                         { entry: updatedSourceEntry.getLogInfo(),
+                           replicationStatus:
+                           updatedSourceEntry.getReplicationStatus(),
+                           reason });
+                return done();
+            }
+            log.error('an error occurred when writing replication ' +
+                      'status',
+                      { entry: updatedSourceEntry.getLogInfo(),
+                        replicationStatus:
+                        updatedSourceEntry.getReplicationStatus() });
+            // Rely on AWS SDK notion of retryable error to decide if
+            // we should retry or give up updating the status.
+            if (err.retryable) {
+                return this._retryUpdateReplicationStatus(
+                    updatedSourceEntry, backoffCtx, log, done);
+            }
```

Comment:
I see `_retryUpdateReplicationStatus(updatedSourceEntry, params, done)` as the function definition. Does `backoffCtx, log` need to be wrapped as an object?